src/dc_context.c

```diff
@@ -407,15 +416,29 @@ static char* get_sys_config_str(const char* key, const char* def)
  */
 int dc_set_config(dc_context_t* context, const char* key, const char* value)
 {
-	int ret = 0;
+	int   ret = 0;
+	char* rel_path = NULL;
 
 	if (context==NULL || context->magic!=DC_CONTEXT_MAGIC || key==NULL) { /* "value" may be NULL */
 		return 0;
 	}
 
-	ret = dc_sqlite3_set_config(context->sql, key, value);
-	update_config_cache(context, key);
+	if (strcmp(key, "selfavatar")==0 && value)
+	{
+		rel_path = dc_strdup(value);
+		if (!dc_make_rel_and_copy(context, &rel_path)) {
+			goto cleanup;
+		}
+		ret = dc_sqlite3_set_config(context->sql, key, rel_path);
+	}
+	else
+	{
+		ret = dc_sqlite3_set_config(context->sql, key, value);
```

Comment:
not sure if it's harder in C wrt GC -- but in general i'd try to normalize this into only one dc_sqlite3_set_config call outside of the if-else-clauses.

Reply:
in general, i agree. but in this case, is think the current code is more clear than having a separate `value_to_set` or so that must not be free()'d. we could also do a `value = rel_path`, but in general i don't like overwriting the given parameters very much (not saying i've not done this here and there)